core/http/src/main/java/org/trellisldp/http/impl/HttpUtils.java

```diff
@@ -393,6 +382,15 @@ public static void closeDataset(final Dataset dataset) {
         }
     }
 
+    /**
+     * Load an Etag generator.
+     * @return the etag generator
+     */
+    public static EtagGenerator loadEtagGenerator() {
```

Comment:
Do we want to bake this into the framework or should we `@Inject` a generator where needed?

Reply:
I don't have a strong opinion -- I basically assume that everyone will use this (default) generator. Adding a different one would simply be a matter of registering it with the service loader, which seems a little lighter-weight than using `@Inject`. Or do you think `@Inject` would be better here?

Part of my logic for using the `ServiceLoader` related to how this gets loaded from "the inside" rather than being registered like the `ResourceService` or `EventService` via the `ServiceBundler`.

Reply:
Also, `@Inject` would require an actual implementation of the interface, while here, we have an anonymous implementation that is created only if another impl isn't available.

Reply:
I'm not too worried about the cost of a single object that lives for the entire application lifetime, and I think the cost of having two different canonical means by which services are wired is much, much higher. If we want to inject, let injection inject.

Reply:
At present, we are using the `ServiceLoader` for the following:

* All Commons-RDF injection (to be removed in #358)
* Injection of `ConstraintRules` in the [http-layer](https://github.com/trellis-ldp/trellis/blob/master/core/http/src/main/java/org/trellisldp/http/impl/BaseLdpHandler.java#L46)
* Injection of `TimemapGenerator` in the [http-layer](https://github.com/trellis-ldp/trellis/blob/master/core/http/src/main/java/org/trellisldp/http/impl/MementoResource.java#L331-L334)
* Injection of `IdentifierService` (throughout)
* Injection of `ActivityStreamService` (in the notification implementations)

Everything else uses plain-old `@Inject`

I can't say that there is a clearly defined distinction between these two categories other than the ServiceLoader-based services tend to be "simpler" (given some definition of "simple").

Perhaps we should merge this change now and address the larger issue of `@Inject` vs. `ServiceLoader` elsewhere?

Reply:
I'm fine with that-- can you put the above list in a ticket and include this new addition?

Reply:
https://github.com/trellis-ldp/trellis/issues/441